### ISSUE TITLE
vmalert panics when -remoteRead.url was specified

### ISSUE THREAD

**Issue by Allineer** [VictoriaMetrics/VictoriaMetrics#586]
**Describe the bug**

vmalert panics on startup when -remoteRead.url was specified:

```
2020-06-26T15:18:42.734Z	info	VictoriaMetrics/app/vmalert/manager.go:83	reading rules configuration file from "/.../alerts.yaml"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6d1d3a]

goroutine 1 [running]:
net/http.(*Transport).roundTrip(0x0, 0xc0000dfb00, 0x185a06df, 0xc, 0xeb350949e449c101)
	net/http/transport.go:485 +0x3a
net/http.(*Transport).RoundTrip(0x0, 0xc0000dfb00, 0x0, 0x0, 0x0)
	net/http/roundtrip.go:17 +0x35
net/http.send(0xc0000dfb00, 0x9a6c20, 0x0, 0x0, 0x0, 0x0, 0xc0000aa478, 0x8f8035, 0x1, 0x0)
	net/http/client.go:252 +0x43e
net/http.(*Client).send(0xc000066270, 0xc0000dfb00, 0x0, 0x0, 0x0, 0xc0000aa478, 0x0, 0x1, 0x6)
	net/http/client.go:176 +0xfa
net/http.(*Client).do(0xc000066270, 0xc0000dfb00, 0x0, 0x0, 0x0)
	net/http/client.go:699 +0x44a
net/http.(*Client).Do(...)
	net/http/client.go:567
github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/datasource.(*VMStorage).Query(0xc000020140, 0x9ae6c0, 0xc000087640, 0xc0001305c0, 0x3c, 0x0, 0x0, 0x0, 0x0, 0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/datasource/vm.go:80 +0x2ef
main.(*AlertingRule).Restore(0xc00016c1b0, 0x9ae6c0, 0xc000087640, 0x9a6480, 0xc000020140, 0x34630b8a000, 0x0, 0xc0001539f0)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/alerting.go:343 +0x1b9
main.(*Group).Restore(0xc0000f6d00, 0x9ae6c0, 0xc000087640, 0x9a6480, 0xc000020140, 0x34630b8a000, 0x880f20, 0xc000153c48)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/group.go:85 +0xf5
main.(*manager).startGroup(0xc0000af8f0, 0x9ae6c0, 0xc000087640, 0xc0000f6d00, 0x1)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/manager.go:67 +0x16c
main.(*manager).update(0xc0000af8f0, 0x9ae6c0, 0xc000087640, 0xc000085440, 0x1, 0x1, 0x10101, 0x0, 0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/manager.go:111 +0x533
main.(*manager).start(...)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/manager.go:52
main.main()
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmalert/main.go:165 +0x847

```

**To Reproduce**

1. vmalert panics with next config:

```
OPTIONS="                                           \
  -httpListenAddr=127.0.0.1:8880                    \
  -datasource.url=http://127.0.0.1:8428             \
  -notifier.url=http://127.0.0.1:9093               \
  -remoteRead.url=http://127.0.0.1:8426             \
  -remoteWrite.url=http://127.0.0.1:8426            \
  -rule=/.../alerts.yaml                            \
"
```

2. vmalert starts without errors with next config: 

```
OPTIONS="                                           \
  -httpListenAddr=127.0.0.1:8880                    \
  -datasource.url=http://127.0.0.1:8428             \
  -notifier.url=http://127.0.0.1:9093               \
  -remoteWrite.url=http://127.0.0.1:8426            \
  -rule=/.../alerts.yaml                            \
"
```

**Version**

```
vmalert-20200625-211530-tags-v1.37.3-0-g423825695
```

**Comment by Allineer** [comment]
PS: vmalert 1.37.0 does not have this error

**Comment by hagen1778** [comment]
Hi @Allineer! The issue supposed to be solved in https://github.com/VictoriaMetrics/VictoriaMetrics/commit/dc4e3f0e0b046db8bbd69bc3e5e9e48795a247ef. 

You can verify it by [building](https://github.com/VictoriaMetrics/VictoriaMetrics/tree/master/app/vmalert#how-to-build-from-sources) `vmalert` from sources. The change will be also included in next release.

**Comment by Allineer** [comment]
`vmalert-20200629-054513-heads-master-0-g82ecfa3b` is working for me.
Thanks, @hagen1778, @kreedom, @tenmozes!

**Comment by valyala** [comment]
Let's re-open the issue until the next release with the bugfix is out.

**Comment by valyala** [comment]
The bugfix has been included in [v1.37.4](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
